samza-azure/src/main/java/org/apache/samza/system/azureblob/avro/AzureBlobAvroWriter.java

```diff
@@ -108,6 +108,8 @@
   private final long maxRecordsPerBlob;
   private final boolean useRandomStringInBlobName;
   private final Object currentDataFileWriterLock = new Object();
+  // when either maxRecordsPerBlob or maxBlobSize is exceeded then a new blob is created
+  // with the same blobURLPrefix and blobNumber is suffixed to the blobURL to differentiate between the blobs
```

Comment:
I thought you used timestamp and a random string to differentiate between blobs. Do you still need `blobNumber`?

Reply:
thanks for pointing out. `blobNumber` was introduced at a time when both timestamp and random string were generated inside AzureBlobSystemProducer. Have removed it now.